src/js/tasks/osSiloPrice.js

```diff
@@ -145,6 +150,182 @@ const calculateMinBuyingPrice = (lendingAPY) => {
     return minPriceScaled > minAllowed ? minPriceScaled : minAllowed;
 };
 
+/**
+ * @notice Estimates the average withdrawal time for a given ARM contract.
+ * @dev This function calculates the estimated withdrawal time based on the availability of tokens
+ *      in the OS Vault and ARM contract, as well as the status of withdrawal requests.
+ * 
+ * Scenarios:
+ * 1. If there is more wS available in the OS Vault than oS in the ARM contract, the withdrawal time
+ *    is estimated to be close to 1 day.
+ * 2. If there are enough unclaimed withdrawal requests older than 13 days to cover the required amount,
+ *    the withdrawal time is estimated to be close to 1 day.
+ * 3. If there are not enough unclaimed withdrawal requests (both older and younger than 13 days) to cover
+ *    the required amount, the withdrawal time is estimated to be the maximum of 14 days.
+ * 4. If there are enough unclaimed withdrawal requests (including recent ones) to cover the required amount,
+ *    the withdrawal time is calculated using a weighted average based on the age of the requests.
+ */
+const estimateAverageWithdrawTime = async (arm, block) => {
+    const blockTag = await getBlock(block);
```

Comment:
nit: probably better to check if the ARM contract was deployed before this block and throw an readable error in that case